core/plugins/org.polarsys.capella.core.sirius.analysis/src/org/polarsys/capella/core/sirius/analysis/TitleBlockServices.java

```diff
@@ -0,0 +1,643 @@
+          if (numCols > 0) {
+            ((DAnnotation) titleBlock).getReferences().add(lineNumber + 2, annotationLine);
+            addColumnsToLine(annotationLine, representation, numCols, lineNumber + 1);
+          }
+        }
+      }
+      renumberColumns((DAnnotation) titleBlock);
+    }
+  }
+
+  private void addColumnsToLine(DAnnotation annotationLine, DRepresentation representation, int numCols,
+      int lineNumber) {
+    List<DAnnotation> annotationCols = new ArrayList<DAnnotation>();
+    for (int j = 0; j < numCols; j++) {
+      DAnnotation annotationCol = DescriptionFactory.eINSTANCE.createDAnnotation();
+      annotationCol.setSource(SOURCE_CELL_PREFIX);
+      annotationCol.getDetails().put(NAME, "");
+      annotationCol.getDetails().put(CONTENT, "");
+      annotationCol.getDetails().put(LINE, String.valueOf(lineNumber));
+      annotationCol.getDetails().put(COLUMN, String.valueOf(j));
+      annotationCol.getDetails().put(IS_ELEMENT_TITLE_BLOCK, TRUE);
+      annotationCols.add(annotationCol);
+      representation.getEAnnotations().add(annotationCol);
+    }
+    annotationLine.getReferences().addAll(annotationCols);
+    representation.getEAnnotations().add(annotationLine);
+  }
+
+  public void createTitleBlockColumn(EObject titleBlock, EObject diagram) {
+    if (!isDiagramTitleBlock((DAnnotation) titleBlock)
+        || "True".equals(((DAnnotation) titleBlock).getDetails().get(IS_ELEMENT_TITLE_BLOCK))) {
+      int columnNumber = Integer.valueOf(((DAnnotation) titleBlock).getDetails().get(COLUMN));
+      titleBlock = getParentTitleBlock((DAnnotation) titleBlock, diagram);
+      DRepresentation representation = null;
+      if ((diagram instanceof DRepresentation)) {
+        representation = (DRepresentation) diagram;
+      }
+      if (representation != null) {
+        if (titleBlock instanceof DAnnotation) {
+          int numLines = ((DAnnotation) titleBlock).getReferences().size();
+          List<EObject> lines = ((DAnnotation) titleBlock).getReferences();
+          for (int i = 0; i < numLines; i++) {
+            DAnnotation annotationCol = DescriptionFactory.eINSTANCE.createDAnnotation();
+            annotationCol.setSource(SOURCE_CELL_PREFIX);
+            annotationCol.getDetails().put(NAME, "");
+            annotationCol.getDetails().put(CONTENT, "");
+            annotationCol.getDetails().put(LINE, String.valueOf(i - 1));
+            annotationCol.getDetails().put(COLUMN, String.valueOf(columnNumber));
+            annotationCol.getDetails().put(IS_ELEMENT_TITLE_BLOCK, TRUE);
+            if (lines.get(i) instanceof DAnnotation) {
+              ((DAnnotation) (lines.get(i))).getReferences().add(columnNumber + 1, annotationCol);
+              representation.getEAnnotations().add(annotationCol);
+            }
+          }
+        }
+      }
+      renumberColumns((DAnnotation) titleBlock);
+    }
+  }
+
+  private void renumberColumns(DAnnotation titleBlock) {
+    int lineNumber = -1;

```

Comment:
why start with -1?

Reply:
I don't know what happens. At some point it adds 1, so I have to start from -1 to number them correctly.